webclient/src/services/CardImporterService.ts

```diff
@@ -0,0 +1,78 @@
+// Fetch and parse card sets
+
+class CardImporterService {
+  importCards(url): Promise<any> {
+    return fetch(url)
+      .then(response => response.json())
+      .then((json) => {
+        const sortedSets = Object.keys(json.data)
+          .map(key => json.data[key])
+          .sort((a, b) => new Date(a.releaseDate).getTime() - new Date(b.releaseDate).getTime());
+
+        const sets = sortedSets.map(({ cards, ...set}) => ({
+          ...set,
+          cards: cards.map(({ name }) => name),
+        }));
+
+        const unsortedCards = sortedSets.reduce((acc, set) => {
+          set.cards.forEach(card => acc[card.name] = card);
+          return acc;
+        }, {});
+
+        const cards = Object.keys(unsortedCards)
+          .sort((a, b) => a.localeCompare(b))
+          .map(key => unsortedCards[key]);
+
+        return { cards, sets };
+      });
+  }
+
+  importTokens(url): Promise<any> {
+    return fetch(url)
+      .then(response => response.text())
+      .then((xmlString) => {
+        const parser = new DOMParser();
+        const dom = parser.parseFromString(xmlString, "application/xml");
+
+        const tokens = Array.from(dom.querySelectorAll('card')).map(
+          (tokenElement) => this.parseXmlAttributes(tokenElement)
+        );
+
+        return tokens;
+      })
+  }
+
+  private parseXmlAttributes(dom: Element) {
```

Comment:
The final data structure here could probably use some work. I'm trying to accommodate the possibility, but not guarantee, that some values like `reverse-related` or `set` might have 1 or more values. Embedding everything within a `{ value: attrValue}` structure is so that we have some consistency when tags might have attribute values, otherwise we end up with an inconsistent structure of strings and complex objects.

This was just my first pass blazing through the import wizard. The final structure can most certainly be improved.